scripts/safari.js

```diff
@@ -7568,287 +7568,280 @@ function Safari() {
+        var betCommands = Object.keys(this.runners).map(function(x) {
+            return link(joinCommand + " " + x + ":" + this.minBet, null, true);
+        }, this);
+        
+        this.betRange = this.silver ? this.minBet + " and " + this.maxBet + " <b>" + itemAlias("silver", true, true) + "s</b>" : "$" + addComma(this.minBet) + " and $" + addComma(this.maxBet);
+        
+        this.joinmsg = "Bet with " + readable(betCommands, "or") + "! Bets must be between " + this.betRange + " (Payout: Favorite " + this.favoritePay + "x, Underdog " + this.underdogPay + "x, Others " + this.normalPay + "x)!";
+        
         sys.sendAll("", safchan);
-        safaribot.sendHtmlAll(sys.name(src) + " is starting a <b>" + this.eventName + "</b> event! The winner will receive <b>" + plural(amount, reward.name) + "</b>!", safchan);
-        safaribot.sendHtmlAll("Type " + link(joinCommand) + " to join (you have " + (this.signupsDuration * this.turnLength) + " seconds)!", safchan);
+        safaribot.sendHtmlAll(sys.name(src) + " is starting a <b>" + this.eventName + "</b> event! The contestants will be " + readable(this.racersList, "and") + ", and they must reach the space " + this.goal + " to win!", safchan);
+        safaribot.sendHtmlAll("Bets must be between " + this.betRange + "! Favorite pays " + this.favoritePay + "x, Underdog pays " + this.underdogPay + "x, others pay " + this.normalPay + "x the bet!", safchan);
+        safaribot.sendHtmlAll("To place your bets, type " + readable(betCommands, "or") + "!", safchan);
         sys.sendAll("", safchan);
     }
-    Marathon.prototype = new SafariEvent();
-    Marathon.prototype.setupEvent = function() {
-        var e, id, len = this.signups.length, player, member, name, names = [];
-        for (e = 0; e < len; e++) {
-            name = this.signups[e];
-            player = getAvatarOff(name);
-            id = player.party[0];
-            member = {
-                owner: name,
-                id: id,
-                position: 0,
-                attackCooldown: 0,
-                hp: sys.baseStats(id, 0),
-                atk: sys.baseStats(id, 1),
-                def: sys.baseStats(id, 2),
-                satk: sys.baseStats(id, 3),
-                sdef: sys.baseStats(id, 4),
-                spd: sys.baseStats(id, 5),
-                damage: 0,
-                movement: 0,
-                lastDamage: 0
-            };
-            this.runners.push(member);
-            this.attackFrequency[name.toLowerCase()] = [];
-            names.push(pokeInfo.icon(member.id) + " " + member.owner + "'s " + poke(member.id));
-        }
-        
-        if (this.signups.length > 6) {
-            this.maxWinners = this.signups.length > 10 ? 3 : 2;
-        } else if (this.signups < 4) {
-            this.hasReward = false;
+    PokeRace.prototype = new SafariEvent();
+    PokeRace.prototype.setupEvent = function() {
+        var e, bet;
+        for (e in this.bets) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.<br>Too many errors. (69% scanned).